python/perf-kernels/layernorm.py

```diff
@@ -7,50 +7,27 @@
 import triton.language as tl
 
 
-def is_cuda():
-    return triton.runtime.driver.active.get_current_target().backend == "cuda"
-
-
-def is_hip():
-    return triton.runtime.driver.active.get_current_target().backend == "hip"
-
-
-def get_cuda_autotune_config():
-    return [
-        triton.Config({}, num_warps=4, num_stages=1),
-        triton.Config({}, num_warps=8, num_stages=1),
-        triton.Config({}, num_warps=16, num_stages=1),
-    ]
-
-
-def get_hip_autotune_config():
+def get_autotune_config():
     return [
-        triton.Config({'waves_per_eu': 1}, num_warps=4, num_stages=1),
-        triton.Config({'waves_per_eu': 1}, num_warps=8, num_stages=1),
-        triton.Config({'waves_per_eu': 1}, num_warps=16, num_stages=1),
-        triton.Config({'waves_per_eu': 2}, num_warps=4, num_stages=1),
-        triton.Config({'waves_per_eu': 2}, num_warps=8, num_stages=1),
-        triton.Config({'waves_per_eu': 2}, num_warps=16, num_stages=1),
-        triton.Config({'waves_per_eu': 4}, num_warps=4, num_stages=1),
-        triton.Config({'waves_per_eu': 4}, num_warps=8, num_stages=1),
-        triton.Config({'waves_per_eu': 4}, num_warps=16, num_stages=1),
+        triton.Config({'waves_per_eu': 1}, num_warps=1),
+        triton.Config({'waves_per_eu': 2}, num_warps=1),
+        triton.Config({'waves_per_eu': 2}, num_warps=2),
+        triton.Config({'waves_per_eu': 1}, num_warps=4),
+        triton.Config({'waves_per_eu': 2}, num_warps=4),
+        triton.Config({'waves_per_eu': 2}, num_warps=8),
     ]
 
 
-def get_autotune_config():
-    if is_cuda():
-        return get_cuda_autotune_config()
-    else:
-        return get_hip_autotune_config()
-
-
 @triton.autotune(configs=get_autotune_config(), key=['n_rows', 'n_cols'], use_cuda_graph=True)
 @triton.jit
-def layernorm_kernel(x_ptr, y_ptr, w_ptr, b_ptr, x_row_stride, y_row_stride, n_rows, n_cols, eps,
-                     BLOCK_SIZE: tl.constexpr):
+def layernorm_kernel_blocked_impl(x_ptr, y_ptr, w_ptr, b_ptr, x_row_stride, y_row_stride, n_rows, n_cols, eps,
+                                  BLOCK_SIZE: tl.constexpr):
 
+    tl.assume(x_row_stride > 0)
+    tl.assume(y_row_stride > 0)
     #program id
     row = tl.program_id(0)
+    tl.assume(row > 0)
```

Comment:
[Question] What `tl.assume` does? Is it a compile time or run time assertion?

[Question] The kernel launch grid if `grid = lambda meta: (n_rows, )`. What's the program ID range? Is it [0, n_rows) (open end) or [1, n_rows] (closed end)? If it is the first option, I believe that we should check for `row >= 0`.